sdk/cosmos/azure-cosmos/azure/cosmos/aio/cosmos_client.py

```diff
@@ -110,7 +110,7 @@ class CosmosClient(object):
     """
 
     def __init__(self, url, credential, consistency_level=None, **kwargs):
-        # type: (str, Any, Optional[str], Any) -> None
+        # type: (str, Any, Optional[str], Any) -> None # pylint: disable=client-accepts-api-version-keyword
```

Comment:
Why not simply accept the api-version kwarg instead of disabling the check?

Reply:
We generally do not provide our users with the ability to choose which API to go against and would instead prefer to ignore this option, since allowing users to pass in this option could lead to more issues for us to debug for no real reason - going against any version that's not the current one is definitely not recommended

Reply:
Is the latest API version guaranteed to be available across all cloud instances at the point where a new client library is released? And what backwards compat guarantees is the service making across api versions?

Reply:
@johanste not sure I understand what you're asking in regards to the SDK - we don't have control over the backend API or what changes they make. If newer versions affect the SDK and breaking changes need to be made, that's a separate issue than the scope of these changes - and only solidifies the fact we don't want users to have access to this setting.

That said, the current version that is to be used for the cosmos python SDK is guaranteed to be available (otherwise it wouldn't be the default).

Reply:
The standard (hence the guideline--hence the pylint rule) is that all Azure SDKs expose api_version as a parameter as a "best effort" option for customers who want to use the latest SDK with an older API version. Forcing customers to use an older version of the SDK to use an older API version would result in us having to hotfix a bunch of old SDKs for bugs whereas this model allows us to recommend that customers use the latest version of the SDK.

The pylint disable statements are there for false alarms. This is not a false alarm and thus it is hard to approve such a change, though I'll definitely defer to @annatisch and @johanste  on this one.

Reply:
@tjprescott @johanste  - This concept does not applies to Cosmos DB at all. Our SDK has more functionality than just serving as a wrapper for the REST calls - we don't support older API versions in our later SDKs. 

Also when we introduce direct mode for the SDK these changes would even become useless for instance as at that point there is no REST API call. 

This is not something that we do in any of our other SDKs.. (Java, .Net, NodeJS, GO SDK)

Reply:
@tjprescott @johanste  - This concept does not applies to Cosmos DB at all. Our SDK has more functionality than just serving as a wrapper for the REST calls - we don't support older API versions in our later SDKs. 

Also when we introduce direct mode for the SDK these changes would even become useless for instance as at that point there is no REST API call. 

This is not something that we do in any of our other SDKs.. (Java, .Net, NodeJS, GO SDK)

Reply:
Agree. Cosmos SDK due to their "thickness" contain logic that is compatible with the current service APIs released in prod. Before any new service API release, the team performs validations that the current existing SDKs will still function properly with any new service release. Some features in the current version in prod on the service might map to some SDK capability that if the customer decides to change the `api_version` would be broken. Breaking the SDK by setting a random lower `api_version` makes little sense to me.
Cosmos DB has no guidance anywhere that guides users to set X or Y `api_version` nor there is any need for such a guidance as far as I understand.